core/templates/pages/exploration-editor-page/services/exploration-rights.service.ts

```diff
@@ -17,206 +17,210 @@
-            data.rights.community_owned, data.rights.viewable_if_private);
-        });
-      },
-      removeRoleAsync: function(memberUsername) {
-        var that = this;
-        var requestUrl = (
-          '/createhandler/rights/' + ExplorationDataService.explorationId);
-
-        return $http.delete(requestUrl, {
-          params: {
-            username: memberUsername
-          }
-        }).then(function(response) {
-          var data = response.data;
-          AlertsService.clearWarnings();
-          that.init(
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import AppConstants from 'assets/constants';
+import { ExplorationDataService } from
+  'pages/exploration-editor-page/services/exploration-data.service';
+import { AlertsService } from 'services/alerts.service';
+import { ExplorationRightsBackendApiService } from './exploration-rights-backend-api.service';
+import { ExplorationRightsBackendData } from './exploration-rights-backend-api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationRightsService {
+  ownerNames: string[];
+  editorNames: string[];
+  voiceArtistNames: string[];
+  viewerNames: string[];
+  private _status: string;
+  private _clonedFrom: string;
+  private _isCommunityOwned: boolean;
+  private _viewableIfPrivate: boolean;
+
+  constructor(
+    private alertsService: AlertsService,
+    private explorationDataService: ExplorationDataService,
+    private explorationRightsBackendApiService:
+      ExplorationRightsBackendApiService
+  ) {}
+
+  init(
+      ownerNames: string[], editorNames: string[], voiceArtistNames: string[],
+      viewerNames: string[], status: string, clonedFrom: string,
+      isCommunityOwned: boolean, viewableIfPrivate: boolean): void {
+    this.ownerNames = ownerNames;
+    this.editorNames = editorNames;
+    this.voiceArtistNames = voiceArtistNames;
+    this.viewerNames = viewerNames;
+    this._status = status;
+    this._clonedFrom = clonedFrom;
+    this._isCommunityOwned = isCommunityOwned;
+    this._viewableIfPrivate = viewableIfPrivate;
+  }
+
+  clonedFrom(): string {
+    return this._clonedFrom;
+  }
+
+  isCloned(): boolean {
+    return Boolean(this._clonedFrom);
+  }
+
+  isPublic(): boolean {
+    return this._status === AppConstants.ACTIVITY_STATUS_PUBLIC;
+  }
+
+  isPrivate(): boolean {
+    return this._status === AppConstants.ACTIVITY_STATUS_PRIVATE;
+  }
+
+  viewableIfPrivate(): boolean {
+    return this._viewableIfPrivate;
+  }
+
+  isCommunityOwned(): boolean {
+    return this._isCommunityOwned;
+  }
+
+  makeCommunityOwned(): Promise<void> {
+    return this.explorationRightsBackendApiService
+      .makeCommunityOwnedPutData(
+        this.explorationDataService.explorationId,
+        this.explorationDataService.data.version, true)
+      .then((response: ExplorationRightsBackendData) => {
+        let data = response;
```

Comment:
Is this line necessary, can't we directly use response object here?

Reply:
No, Removed ( Done )